tools/swift-format/Sources/Rules/MaximumBlankLines.swift

```diff
@@ -16,5 +16,69 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#vertical-whitespace
 public final class MaximumBlankLines: SyntaxFormatRule {
+  public override func visit(_ token: TokenSyntax) -> Syntax {
+    guard let parentToken = token.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
 
+    guard let grandparentTok = parentToken.parent else {
+      return token.withLeadingTrivia(removeExtraBlankLines(token).0)
+    }
+
+    // Tokens who appeared in a member type are handle by
+    // BlankLineBetweenMembers rule.
+    if grandparentTok is MemberDeclListItemSyntax {
+      return token
+    }
+    else {
+      let correctTrivia = removeExtraBlankLines(token)
+      return correctTrivia.1 ? token : token.withLeadingTrivia(removeExtraBlankLines(token).0)
```

Comment:
It looks like the call to `removeExtraBlankLines(token)` on this line is unnecessary, since you call it above. So this could be `token.withLeadingTrivia(correctTrivia)`?